subsys/bluetooth/host/hci_core.c

```diff
@@ -3736,6 +3736,10 @@ int bt_disable(void)
 	/* Clear BT_DEV_ENABLE here to prevent early bt_enable() calls */
 	atomic_clear_bit(bt_dev.flags, BT_DEV_ENABLE);
 
+#if defined(CONFIG_BT_PER_ADV_SYNC)
+	bt_periodic_sync_disable();
+#endif /* CONFIG_BT_PER_ADV_SYNC */
```

Comment:
It sort of feels like this should be done before `atomic_clear_bit(bt_dev.flags, BT_DEV_ENABLE);`

Reply:
Moved up and extended comment related with clear of `BT_DEV_ENABLE` bit.